src/docs/cmdstan-guide/json-format.tex

```diff
@@ -0,0 +1,111 @@
+\chapter{JSON Format}\label{json.chapter}
+
+\noindent 
+\CmdStan can use JSON format for input data for both
+model data and parameters. Model data is read in by the model
+constructor. Model parameters are used to initialize the sampler and
+optimizer.
+
+\section{Stan Data Types in JSON Notation}
+
+A Stan input file in  JSON notation consists of  JSON object which contains zero
+or more name-value pairs.  This structure corresponds to a Python data
+dictionary object.  The following is an example of JSON data for the
+simple Bernoulli example model:
+
+%
+\begin{quote}
+\begin{Verbatim}
+{ "N" : 10, "y" : [ 0, 1, 0, 0, 0, 0, 0, 0, 0, 1 ] }
+\end{Verbatim}
+\end{quote}
+%
+
+Matrix data and multi-dimensional arrays are indexed in row-major
+order.  For a Stan program which has data block
+\begin{quote}
+\begin{Verbatim}
+data {
+  int d1;
+  int d2;
+  int d3;
+  int ar[d1, d2, d3];
+}
+\end{Verbatim}
+\end{quote}
+\noindent
+the following JSON input file would be valid
+\begin{quote}
+\begin{Verbatim}
+{
+    "d1": 2,
+    "d2" : 3,
+    "d3" : 4,
+    "ar": [[[0, 1, 2, 3], [4, 5, 6, 7], [8, 9, 10, 11]], 
+            [[12, 13, 14, 15], [16, 17, 18, 19], [20, 21, 22, 23]]]
+}
+\end{Verbatim}
+\end{quote}
+
```

Comment:
Could you indicate what the spacing requirements are for JSON---is whitespace always ignored?  If so, should we be encouraging people to use more compressed versions of these with fewer whitespaces?

Presumably the differences in `"d1":` and `"d2" :` spacing were intentional.  If so, please call out that they were written differently on purpose?

Reply:
done